test/reporter/test-reporter-tap.js

```diff
@@ -70,6 +74,21 @@ test('reporter.tap(): failing', function (t) {
   t.end();
 });
 
+test('reporter.tap(): parser', function (t) {
+  var output = '';
+  function logger(message) {
+    output += message;
+  }
+
+  tap(logger, failingInput);
+  var p = parser(function (results) {
+    var expected = JSON.stringify(results, null, 2) + '\n';
```

Comment:
if change tapParserExpected to be a json object as mentioned above we can use t.deepEquals to compare the two and avoid the JSON.stringify